common/config/src/main/java/io/helidon/common/config/GlobalConfig.java

```diff
@@ -47,6 +47,7 @@ public final class GlobalConfig {
     private static final System.Logger LOGGER = System.getLogger(GlobalConfig.class.getName());
     private static final AtomicBoolean LOGGED_REGISTERED = new AtomicBoolean(false);
     private static final Config EMPTY = Config.empty();
+    private static final AtomicReference<Boolean> GLOBAL_FROM_REGISTRY = new AtomicReference<>();
```

Comment:
Could be AtomicBoolean or even just primitive volatile boolean

Reply:
I needed three states (null/true/false) originally and then refactored it to just need two. Will use AtomicBoolean